### PR TITLE
Fix PaperPluginMeta#getDisplayName

### LINKED ISSUE #9027: PaperPluginMeta#getDisplayName does not match its documented behavior by excluding the version
### Expected behavior

Either:
- `PaperPluginMeta#getDisplayName` returns a string that includes the plugin version.
- The documentation for `PluginMeta#getDisplayName` doesn't include any mention of version.

### Observed/Actual behavior

The documentation for `PluginMeta#getDisplayName` specifies that the plugin version is included in the returned string:

> Returns the display name of the plugin, *including the version*.
>
> **Returns:**
> a descriptive name of the plugin *and respective version*

Calling `PaperPluginMeta#getDisplayName` does not include the version, but rather just the name of the plugin with underscores replaced with spaces. This can be confirmed by looking at the source code of `PaperPluginMeta`:

(line 110, in `static PaperPluginMeta create(BufferedReader reader)`, with `pluginConfiguration` being the `PaperPluginMeta` being constructed)
```java
pluginConfiguration.displayName = pluginConfiguration.name.replace('_', ' ');
```

(lines 130-133)
```java
@Override
public @NotNull String getDisplayName() {
    return this.displayName;
}
```

### Steps/models to reproduce

`paper-plugin.yml`:
```yaml
name: Test Plugin
version: '1.0'
main: TestPlugin
api-version: '1.19'
bootstrapper: TestBootstrap
```

`TestPlugin.java`:
```java
public class TestPlugin extends JavaPlugin {
}
```

`TestBootstrap.java`:
```java
public class TestBootstrap implements PluginBootstrap {
    @Override
    public void bootstrap(@NotNull PluginProviderContext context) {
        context.getLogger().info(context.getConfiguration().getDisplayName());
    }
}
```

Observe the logged display name when the bootstrapper is invoked. If the incorrect behavior is present, `Test Plugin` will be logged. Otherwise, `Test Plugin v1.0` or similar will be logged.

### Plugin and Datapack List

Plugins:
* An in-development Paper (i.e. `paper-plugin.yml`) plugin

Data packs:
```
[17:18:05 INFO]: There are 2 data packs enabled: [vanilla (built-in)], [file/bukkit (world)]
[17:18:05 INFO]: There are no more data packs available
```

### Paper version

```
[17:16:58 INFO]: This server is running Paper version git-Paper-466 (MC: 1.19.4) (Implementing API version 1.19.4-R0.1-SNAPSHOT) (Git: 7af4cd3)
You are running the latest version
Previous version: git-Paper-463 (MC: 1.19.4)
```

### Other

_No response_

### PR DESCRIPTION
**Pull request by TheRealRyGuy**
fixes #9027, below is the only change
```java
@Override
    public @NotNull String getDisplayName() {
        return this.displayName + " v" + this.getVersion();
    }
```

### CODE REVIEW

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
The fix for this is to change what is assigned to the `displayName` field when the meta is created.

**Comment by Owen1212055** [comment]
Also please merge this change into the original paper plugin patch.

**Comment by TheRealRyGuy** [comment]
```java
pluginConfiguration.displayName = pluginConfiguration.name.replace('_', ' ') + " v" + pluginConfiguration.version;
```
Assuming the original patch was `patches/server/0013-Paper-Plugins.patch`, done

**Review by Machine-Maker — APPROVED** [review]

**Comment by TheLukeGuy** [comment]
Thanks for fixing this. :)

The behavior of the `/version` command with just this change is a little weird with Paper plugins:
```
> pl
[21:36:12 INFO]: Server Plugins (2):
[21:36:12 INFO]: Paper Plugins:
[21:36:12 INFO]:  - Outlet
[21:36:12 INFO]: Bukkit Plugins:
[21:36:12 INFO]:  - Essentials
> ver outlet
[21:36:13 INFO]: Outlet v1.0.0-SNAPSHOT version 1.0.0-SNAPSHOT
[21:36:13 INFO]: A modern plugin manager for Paper.
[21:36:13 INFO]: Website: https://github.com/TheLukeGuy/outlet
[21:36:13 INFO]: Author: TheLukeGuy
> ver essentials
[21:36:15 INFO]: Essentials version 2.20.0-dev+45-bf14b88
[21:36:15 INFO]: Provides an essential, core set of commands for Bukkit.
[21:36:15 INFO]: Website: https://essentialsx.net/
[21:36:15 INFO]: Authors: Zenexer, ementalo, Aelux, Brettflan, KimKandor, snowleo, ceulemans, Xeology, KHobbits, md_5, Iaccidentally, drtshock, vemacs, SupaHam, mdcfe, JRoy and pop4959
```

As you can see, the first line of `/ver [paper plugin name]` is `[Plugin name] v[plugin version] version [plugin version]`, while the first line of `/ver [bukkit plugin name]` is `[Plugin name] version [plugin version]`. This was the only oddity that I could find from a couple minutes of basic testing, but I didn't do a deep look into where `getDisplayName` is used.

**Comment by TheRealRyGuy** [comment]
Should've fixed version command, didn't realize it relied on description files 
This did modify PluginDescriptionFile instantiation which probably has repercussions I can't think of, but here we are. Current application makes more sense in my head, as `PluginDescriptionFile#rawName` should have underscores, where `#name` should be the formatted nice name I feel like

File: patches/server/0013-Paper-Plugins.patch [hunk 1]
```diff
@@ -960,8 +960,8 @@ index 0000000000000000000000000000000000000000..ecd11d3bd6c7e80a9b33cb8c2d7ed8a5
 +    public void init(JavaPlugin plugin) {
 +        PluginMeta config = this.configuration;
 +        PluginDescriptionFile pluginDescriptionFile = new PluginDescriptionFile(
-+            config.getName(),
 +            config.getDisplayName(),
++            config.getName(),
```
Comment (Machine-Maker) [inline patches/server/0013-Paper-Plugins.patch]: I disagree with this change. the rawName and name parameters should be set correctly. the `displayName` is not correct here. The first parameter should just be getName(), the second parameter should be getName() but replacing `_` with ` ` (which is what getDisplayName() used to be).

**Review by Machine-Maker — CHANGES_REQUESTED** [review]

**Comment by TheRealRyGuy** [comment]
yeah you right, that made no sense oops
this (should?) be the final major place where versioning needs a quick fix though

**Review by Machine-Maker — APPROVED** [review]

**Review by Owen1212055 — APPROVED** [review]

**Pull request merged**